modulegen/main.go

```diff
@@ -1,76 +1,76 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
 	"path/filepath"
 
+	"github.com/spf13/cobra"
+
 	"github.com/testcontainers/testcontainers-go/modulegen/internal/tools"
 )
 
-var (
-	asModuleVar  bool
-	nameVar      string
-	nameTitleVar string
-	imageVar     string
-)
+func NewRootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "modulegen",
+		Short: "Management tool for testcontainers-go",
+		Long:  "Management tool for testcontainers-go",
+	}
+	cmd.AddCommand(NewCmd())
+	return cmd
+}
 
-func init() {
-	flag.StringVar(&nameVar, "name", "", "Name of the example. Only alphabetical characters are allowed.")
-	flag.StringVar(&nameTitleVar, "title", "", "(Optional) Title of the example name, used to override the name in the case of mixed casing (Mongodb -> MongoDB). Use camel-case when needed. Only alphabetical characters are allowed.")
-	flag.StringVar(&imageVar, "image", "", "Fully-qualified name of the Docker image to be used by the example")
-	flag.BoolVar(&asModuleVar, "as-module", false, "If set, the example will be generated as a Go module, under the modules directory. Otherwise, it will be generated as a subdirectory of the examples directory.")
+func NewCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "new",
+		Short: "Create a new Example or Module",
+		Long:  "Create a new Example or Module",
+	}
+	cmd.AddCommand(NewExampleCmd())
+	cmd.AddCommand(NewModuleCmd())
+	return cmd
 }
 
 func main() {
-	required := []string{"name", "image"}
-	flag.Parse()
-
-	seen := make(map[string]bool)
-	flag.Visit(func(f *flag.Flag) { seen[f.Name] = true })
-	for _, req := range required {
-		if !seen[req] {
-			// or possibly use `log.Fatalf` instead of:
-			fmt.Fprintf(os.Stderr, "missing required -%s argument/flag\n", req)
-			os.Exit(2) // the same exit code flag.Parse uses
-		}
+	err := NewRootCmd().Execute()
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
 	}
+}
 
+func generateExample(exampleVar *ExampleVar, isModule bool) error {
```

Comment:
Let's be intentional here and call it:
```suggestion
func generateModule(exampleVar *ExampleVar, asExample bool) error {
```

We can rename the Example struct to `TCModule` (or similar, to avoid just `Module`) here or in a follow-up.

Reply:
Also: do we need pointer semantics here for the `exampleVar`? We are not modifying its state in this method, as we only use it to instantiate the Example struct, right?

Reply:
No modification indeed. I removed the pointers

Reply:
> We can rename the Example struct to TCModule (or similar, to avoid just Module) here or in a follow-up.

Let's do it in a follo-up, yes.